routers/algorithm.py

```diff
@@ -0,0 +1,81 @@
+import pandas as pd
+from supabase import create_client, Client
+import requests
+import numpy as np
+import routingpy as rp
+from python_tsp.exact import solve_tsp_dynamic_programming
+import os
+from fastapi import APIRouter, Depends
+from models import supabase, User
+from shapely import wkb
+
+router = APIRouter(prefix="/algorithm", tags=["Algorithm"])
+
+graphhopper_api_key = os.environ.get("GRAPHOPPER_API_KEY", None)
```

Comment:
We need to pass this in deployment scripts and set the values in the GitHub secrets. Might also be done as a follow-up PR, no strong preference

Reply:
Please set the GRAPHOPPER_API_KEY value in the Github